src/System.Private.CoreLib/shared/System/Reflection/MethodInfo.cs

```diff
@@ -13,15 +13,15 @@ public abstract partial class MethodInfo : MethodBase
         public override MemberTypes MemberType => MemberTypes.Method;
 
         public virtual ParameterInfo? ReturnParameter { get { throw NotImplemented.ByDesign; } }
```

Comment:
Since we have changed `ReturnTypeCustomAttributes` to be non-null (ignoring the buggy MethodInfoBuilder implementation), I think we should change this one to be non-null as well.

Reply:
RTDynamicMethod returns null directly on its override, I think we should leave the virtual as nullable as well to express that, no?

Reply:
RTDynamicMethod should create the ParameterInfo for return value same way as it creates one for the regular parameters. It is a bug that should be fixed.

Reply:
Fixed.